edge-hub/core/test/Microsoft.Azure.Devices.Edge.Hub.E2E.Test/ProtocolHeadFixture.cs

```diff
@@ -78,7 +78,19 @@ async Task StartProtocolHead()
             // TODO - After IoTHub supports MQTT, remove this and move to using MQTT for upstream connections
             await ConnectToIotHub(edgeDeviceConnectionString);
 
-            ConfigHelper.TestConfig[EdgeHubConstants.ConfigKey.IotHubConnectionString] = edgeDeviceConnectionString;
+            // Set edgeHub connection string to config
+            string iotHubConnectionString = await SecretsHelper.GetSecretFromConfigKey("iotHubConnStrKey");
+            string edgeDeviceId = ConnectionStringHelper.GetDeviceId(edgeDeviceConnectionString);
+            string iothub = ConnectionStringHelper.GetHostName(edgeDeviceConnectionString);
+            RegistryManager rm = RegistryManager.CreateFromConnectionString(iotHubConnectionString);
+            var edgeHubModule = await rm.GetModuleAsync(edgeDeviceId, "$edgeHub");
+            if (edgeHubModule.Authentication.Type == AuthenticationType.None)
+            {
+                edgeHubModule.Authentication.Type = AuthenticationType.Sas;
+                edgeHubModule = await rm.UpdateModuleAsync(edgeHubModule);
+            }
+
+            ConfigHelper.TestConfig[EdgeHubConstants.ConfigKey.IotHubConnectionString] = $"HostName={iothub};DeviceId={edgeDeviceId};ModuleId=$edgeHub;SharedAccessKey={edgeHubModule.Authentication.SymmetricKey.PrimaryKey}";
```

Comment:
How is this connection string we are setting different from the ```edgeDeviceConnectionString``` we get from keyvault above?

Reply:
keyvault has device connection string and here it's using edgeHub connection string